src/platform/intel/cavs/lib/power_down.S

```diff
@@ -132,12 +139,12 @@ _PD_SEND_IPC:
  */
 	movi temp_reg0, IPC_HOST_BASE
 	l32i temp_reg1, temp_reg0, IPC_DIPCTDR
-	movi temp_reg2, IPC_DIPCTDR_BUSY
+	movi temp_reg2, ipc_flag
+	l32i temp_reg2, temp_reg2, 0
 	or temp_reg1, temp_reg1, temp_reg2
 	s32i temp_reg1, temp_reg0, IPC_DIPCTDR
 
 	l32i temp_reg1, temp_reg0, IPC_DIPCTDA
-	movi temp_reg2, IPC_DIPCTDA_DONE
```

Comment:
@mmaka1 the IPC_DIPCTDA_DONE setting is missed with this change?

Reply:
No, 0x80000000 is loaded to a8 earlier to clear the busy interrupt and there is no need to load the same value again. Note that all tests that execute ctx_save would fail w/o DA's DONE set.

Reply:
@mmaka1 Oh that's true. And the IPC_DIPCTDR_BUSY is already cleared also, shouldn't we remove the whole _PD_SEND_IPC part here?

```
	/* write 1 to clear busy, and trigger interrupt to host*/
#if CAVS_VERSION == CAVS_VERSION_1_5
	ipc_write(IPC_DIPCT, ipc_read(IPC_DIPCT) | IPC_DIPCT_BUSY);
#else
	ipc_write(IPC_DIPCTDR, ipc_read(IPC_DIPCTDR) | IPC_DIPCTDR_BUSY);
	ipc_write(IPC_DIPCTDA, ipc_read(IPC_DIPCTDA) | IPC_DIPCTDA_DONE);
#endif
```

Reply:
Both registers are still programmed in the _PD_SEND_IPC, I just removed reloading a8 with the same const value not the register programming, and that programming is required. The code you referred to is on a different path.